### PR TITLE
[core] time travel add datetime format

### PR DESCRIPTION
**Pull request by seawenc**
### Purpose

Linked issue: https://github.com/apache/incubator-paimon/issues/2475

### Tests
unit test is paas

integration testing： 

#### flink test
image

#### hive test

image

### CODE REVIEW

File: docs/content/how-to/querying-tables.md [hunk 1]
```diff
@@ -51,6 +51,11 @@ SELECT * FROM t /*+ OPTIONS('scan.snapshot-id' = '1') */;
 -- read the snapshot from specified timestamp in unix milliseconds
 SELECT * FROM t /*+ OPTIONS('scan.timestamp-millis' = '1678883047356') */;
 
+-- read the snapshot from specified datetime
+-- Supported formats include：yyyy,yyyy-MM,yyyy-MM-dd, yyyy-MM-dd HH, yyyy-MM-dd HH:mm, yyyy-MM-dd HH:mm:ss, yyyy-MM-dd HH:mm:ss.SSS
+-- default local time zone,If you need to specify a time zone, please end with '/UTC{offset}',for example: 2023-12-11 12:12/UTC+8
+SELECT * FROM t /*+ OPTIONS('scan.datetime' = '2023-12-09 23:09') */;
```
Comment (JingsongLi) [inline docs/content/how-to/querying-tables.md:57]: just `scan.timestamp`?

Comment (seawenc) [reply docs/content/how-to/querying-tables.md]: Has been modified to `scan.timestamp` ,please recheck!

image

**Comment by seawenc** [comment]
@JingsongLi  Please recheck!

File: paimon-common/src/main/java/org/apache/paimon/utils/DateTimeUtils.java [hunk 2]
```diff
@@ -76,6 +84,95 @@ public class DateTimeUtils {
...
+    };
+    private static final String FULL_TIMESTAMP_DEF = "0000-01-01 00:00:00.000";
+
+    /**
+     * auto format date.
+     *
+     * @param timestampString
+     * @return
+     */
+    public static Timestamp autoFormatToTimestamp(String timestampString) {
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/utils/DateTimeUtils.java:106]: Can we just use `parseTimestampData`? It is very confused to introduce another parser here.

Comment (seawenc) [reply paimon-common/src/main/java/org/apache/paimon/utils/DateTimeUtils.java]: @JingsongLi  Fixed，Please recheck!

**Comment by JingsongLi** [comment]
+1 thanks @seawenc for the contribution!

**Pull request merged**